src/libs/installer/packagemanagercore.cpp

```diff
@@ -648,6 +648,20 @@ int PackageManagerCore::downloadNeededArchives(double partProgressSize)
     ProgressCoordinator::instance()->registerPartProgress(&archivesJob,
         SIGNAL(progressChanged(double)), partProgressSize);
 
+    // Print some progress information on console as well
+    connect(&archivesJob, &DownloadArchivesJob::outputTextChanged, [](const QString &progress) {
+        qDebug().noquote() << progress;
+    });
+    connect(&archivesJob, &DownloadArchivesJob::progressChanged,
+            [lastReported = -10, time = QTime::currentTime()](double progress) mutable {
+        int roughProgress = static_cast<int>(progress * 10) * 10;
+        if (roughProgress > lastReported && time.elapsed() > 3000 || time.elapsed() > 15000) {
+            qDebug().nospace() << qMax(roughProgress, 1) << "% ...";
```

Comment:
What's the point of qMax here? Why can't we report 0%?

Reply:
It would look odd (and less accurate)